openedx/core/djangoapps/course_roles/helpers.py

```diff
@@ -0,0 +1,189 @@
+    """
+    Returns ture if permissions checks should be used
+    """
+    return USE_PERMISSION_CHECKS_FLAG.is_enabled()
+
+
+@request_cached()
+def user_has_permission_course(
+        user: Union[User, AnonymousUser],
+        permission: Union[CourseRolesPermission, str],
+        course_key: CourseKey
+):
+    """
+    Check if a user has a permission in a course.
+    """
+    if not use_permission_checks():
+        return False
+    elif isinstance(user, AnonymousUser):
+        return False
+    if isinstance(permission, CourseRolesPermission):
+        permission = permission.value.name
+    return UserRole.objects.filter(
+        user=user,
+        role__permissions__name=permission,
+        course=course_key,
+    ).exists()
+
+
+@request_cached()
+def user_has_permission_list_course(
+        user: Union[User, AnonymousUser],
+        permissions: List[Union[CourseRolesPermission, str]],
+        course_key: CourseKey
+):
+    """
+    Check if a user has all of the given permissions in a course.
+    """
+    if not use_permission_checks():
+        return False
+    return all(user_has_permission_course(user, permission, course_key) for permission in permissions)
+
+
+@request_cached()
+def user_has_permission_list_any_course(
+        user: Union[User, AnonymousUser],
+        permissions: List[Union[CourseRolesPermission, str]],
+        course_key: CourseKey
+):
+    """
+    Check if a user has ANY of the given permissions in a course.
+    """
+    return any(user_has_permission_course(user, permission, course_key) for permission in permissions)
+
+
+@request_cached()
+def user_has_permission_org(
+        user: Union[User, AnonymousUser],
+        permission: Union[CourseRolesPermission, str],
+        organization_name: str
+):
+    """
+    Check if a user has a permission for all courses in an organization.
+    """
+    if not use_permission_checks():
+        return False
+    elif isinstance(user, AnonymousUser):
+        return False
+    if isinstance(permission, CourseRolesPermission):
+        permission = permission.value.name
+    return UserRole.objects.filter(
+        user=user,
+        role__permissions__name=permission,
+        course__isnull=True,
+        org__name=organization_name,
+    ).exists()
+
+
+@request_cached()
+def user_has_permission_list_org(
+        user: Union[User, AnonymousUser],
+        permissions: List[Union[CourseRolesPermission, str]],
+        organization_name: str
+):
+    """
+    Check if a user has ALL of the given permissions for all courses in an organization.
+    """
+    if not use_permission_checks():
+        return False
+    return all(
+        user_has_permission_org(user, permission, organization_name) for permission in permissions
+    )
+
+
+@request_cached()
+def user_has_permission_course_org(
+        user: Union[User, AnonymousUser],
+        permission: Union[CourseRolesPermission, str],
+        course_key: CourseKey
+):
```

Comment:
If this is the most commonly called one, could it be called `user_has_permission`? It's the `course_roles` app, so between that and the `course_key` argument, it should be clear that it's scoped to a course.